app/controllers/application_controller.rb

```diff
@@ -1,2 +1,9 @@
 class ApplicationController < ActionController::Base
+  before_action :set_current_user
+
+  private
+
+  def set_current_user
+    AdminUser.current_admin_user = current_admin_user
```

Comment:
I saw I think something new in Rails 5.2 https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html
From what I read this does exactly what you need, but without an extra gem. Maybe it's worth a try?

Reply:
Alright, I haven't heard about this =D I'll take a look at it tomorrow! =)

Reply:
This is exactly what we need. =) I've updated the code to use this instead of the request_store. I seem to still need to clear the Current.admin_user on the before(:each) of the litigations controller spec, otherwise my specs fail :/

Basically I'd expect the before(:each) to always sign in the new admin_user, but it seems that we need clear it by hand or maybe we could log it out. Any thoughts?

Cheers!

Reply:
I will look at the specs.

Reply:
Crap, I have no idea why is that not removed, but maybe we can add global clear of all current attributes after each controller test?

```
  config.after :each, type: :controller do
    ActiveSupport::CurrentAttributes.reset_all
  end
```

Reply:
in rails_helper

Reply:
I think controller specs are not just invoking the actual reset code so that could be a good workaround. Maybe it's worth to look at request specs instead of controller.

Reply:
oh right, that might be it. I'll add your thing to the rails_helper for the moment. Spent too much time on this already :p